Src/Support/GoogleApis.Auth.PlatformServices_Shared/OAuth2/Pkcs8.cs

```diff
@@ -0,0 +1,266 @@
+                    {
+                        throw new InvalidOperationException("Invalid Oid, too long.");
+                    }
+                    List<int> result = new List<int>();
+                    bool first = true;
+                    while (_index < endOffset)
+                    {
+                        int subId = 0;
+                        byte b;
+                        do
+                        {
+                            b = NextByte();
+                            if ((subId & 0xff000000) != 0)
+                            {
+                                throw new NotSupportedException("Oid subId > 2^31 not supported.");
+                            }
+                            subId = (subId << 7) | (b & 0x7f);
+                        } while ((b & 0x80) != 0);
+                        if (first)
+                        {
+                            first = false;
+                            result.Add(subId / 40);
+                            result.Add(subId % 40);
+                        }
+                        else
+                        {
+                            result.Add(subId);
+                        }
+                    }
+                    return result.ToArray();
+                }
+
+                private object[] ReadSequence()
+                {
+                    int length = ReadLength();
+                    int endOffset = _index + length;
+                    if (endOffset < 0 || endOffset > _bs.Length)
+                    {
+                        throw new InvalidOperationException("Invalid sequence, too long.");
+                    }
+                    List<object> sequence = new List<object>();
+                    while (_index < endOffset)
+                    {
+                        sequence.Add(Decode());
+                    }
+                    return sequence.ToArray();
+                }
+
+                private byte NextByte() => _bs[_index++];
+
+                public byte[] ReadBytes(int length)
+                {
+                    if (length <= 0)
+                    {
+                        throw new ArgumentOutOfRangeException(nameof(length), "length must be positive.");
+                    }
+                    if (_bs.Length - length < 0)
+                    {
+                        throw new ArgumentException("Cannot read past end of buffer.");
+                    }
+                    byte[] result = new byte[length];
+                    Array.Copy(_bs, _index, result, 0, length);
+                    _index += length;
+                    return result;
+                }
+
+                public Tag ReadTag()
+                {
+                    byte b = NextByte();
+                    int tag = b & 0x1f;
+                    if (tag == 0x1f)
+                    {
+                        throw new NotSupportedException("Tags of value > 30 not supported.");
```

Comment:
This is a bit confusing - would 0x1f indicate a continuation? The condition doesn't match the message, basically. It may well be that the code is fine, but it could do with a comment explaining it.